src/hdmf/backends/hdf5/h5tools.py

```diff
@@ -344,6 +345,33 @@ def read_builder(self):
             self.__read[self.__file] = f_builder
         return f_builder
 
+    def __set_written(self, builder):
+        """
+        Mark this builder as written.
+
+        :param builder: Builder object to be marked as written
+        :type builder: Builder
+        """
+        # currently all values in self._written_builders are True, so this could be a set but is a dict for
+        # future flexibility
+        builder_id = self.__bldrhash__(builder)
+        self._written_builders[builder_id] = True
+
+    def get_written(self, builder):
+        """Return True if this builder has been written to (or read from) disk by this IO object, False otherwise.
+
+        :param builder: Builder object to get the written flag for
+        :type builder: Builder
+
+        :return: True if the builder is found in self._written_builders using the builder ID, False otherwise
+        """
+        builder_id = self.__bldrhash__(builder)
+        return self._written_builders.get(builder_id, False)
+
+    def __bldrhash__(self, obj):
```

Comment:
I'd suggest to rename to ``__builderhash``

Reply:
Done. (objectmapper.py has a function `__bldrhash__` which is where I copied this from.)